### DISCUSSION TITLE
Overlay Stance Resetting to Default upon Montage Play

### DISCUSSION THREAD

**Discussion by Merkatroid1** [Sixze/ALS-Refactored#196] [Q&A]
Hey Sixze,
Might be a stupid issue, but I followed your previous guide here and for the most part everything works quite well.
https://github.com/Sixze/ALS-Refactored/discussions/19

The only issue is that the stance (M4 -> Aiming) is defaulting to the upright pose instead of remaining in the Aiming pose. 
Any ideas for it to remain in the Aiming pose (stop character from standing upright)?
https://streamable.com/ktmvp2

For some more info - only the left and right arms as well as the head have layering curves applied (1) - everything else is at 0.

**Comment by Sixze** [comment]
> everything else is at 0

This is the answer. Сurves control the blending of the overlay pose with the locomotion pose, and if they are all 0 then the default idle pose coming from the locomotion will be used.

**Reply by Merkatroid1** [reply]
Thanks Sixze - working pretty much perfectly now.
https://streamable.com/df600t

Just had to play around with the curves some more - particularly the additive curves.
Additionally, in the montage I didn't need to add animations for the spine/pelvis/legs layers but still needed to specify the layering curves as 1 in the anim.

**Reply by DBTech77** [reply]
@Merkatroid1 I'm curious which curves in particular you modified to get it to work properly? I'm facing this same issue, the reload montage works properly if the character is moving, but if they're standing still the legs will move to the reload animation position and then back to the default position

**Reply by Merkatroid1** [reply]
@DBTech77 Heyo
Add **ALL** montage layers - in all you should have:
Head, ArmLeft, ArmRight, Spine, Pelvis, Legs, Curves
Even if they're blank, you still need the layers there

E.g. 
![20221213_013937](https://user-images.githubusercontent.com/111747271/207212933-b8eefe4b-73f0-4f45-a7ff-427f4f28ef24.png)

I've been doing some testing and here's what I've come up with so far - will update this post upon further testing.

Now we have 3 items of interest
1) The anim curves (ANIM)
2) The anim slots (ANIM)
3) The montage layers (MONTAGE)

For anim curves, 0 = default overlay, 1 = current/set overlay
For anim slots, 0 = no curve, 1 = full curve
<<  The anim slot modifies the anim curve e.g. slot 0.45 = 45% of corresponding curve (55% default overlay) >>
<<  curve values CANNOT be modified, it's either enabled or disabled  >>
For montage layers, NONE = current overlay, FILLED = set overlay

**Reply by DBTech77** [reply]
@Merkatroid1 This is a great explanation thank you! I was at a loss as to what the curves value changed, I finally figured out my issue, I didn't have the layering slot curves, I only had the base layering curves so the slots were being ignored in the main anim bp.

For anyone else encountering this issue just make sure you have 2 sequences, 1 with curves and 1 without, the sequence with curves should have a total of 25 curves, 6 slot and 19 layering.

Then just follow Merkatroid1's explanation above to set the curve values to something that works for the specific sequences you're using

![Untitled](https://user-images.githubusercontent.com/83677055/207215482-032f7b62-4523-4291-95d2-84f9f16f5aa6.jpg)

**Comment by Merkatroid1** [comment]
Update to the post.
See the possible combinations below:

![20221213_053425](https://user-images.githubusercontent.com/111747271/207243851-eb7236dc-caac-4560-95d1-6e67fac7ce87.png)

If anything is incorrect here feel free to correct me and I'll edit it.

**Reply by noreply13** [reply]
"I did everything as written above. But the montage starts not from the current 'Rifle' layer for some reason, but from 'Default' layer and then switches abruptly. As a result, I see jerks at the beginning and at the end. And at the end of the montage, it returns the hands to the 'default' layer, then suddenly moves to the needed layer. Where is the problem?"
